tests/unit/request/xhr.ts

```diff
@@ -2,168 +2,497 @@ import * as registerSuite from 'intern!object';
+				function (response: any) {
+					const query = JSON.parse(response.data).query;
+					assert.strictEqual(query.color, 'blue');
+					assert.deepEqual(query.numbers, [ 'one', 'two' ]);
+					assert.strictEqual(
+						response.url,
+						'/__echo/xhr?color=blue&numbers=one&numbers=two'
+					);
+				}
+			);
+		},
+
+		'.post': function () {
+			if (!echoServerAvailable) {
+				this.skip('No echo server available');
+			}
+			return xhrRequest('/__echo/post', {
+				method: 'POST',
+				data: new UrlSearchParams({ color: 'blue' }).toString(),
+				responseType: 'json'
+			}).then(
+				function (response: any) {
+					const data = JSON.parse(response.data);
+					assert.strictEqual(data.method, 'POST');
+					const payload = data.payload;
+
+					assert.ok(payload && payload.color);
+					assert.strictEqual(payload.color, 'blue');
+				}
+			);
 		}
 	},
 
 	'request options': {
-		'timeout'(): void {
-			const dfd = this.async();
-			xhrRequest(getRequestUrl('foo.json'), { timeout: 1 })
+		'timeout'() {
+			if (!echoServerAvailable) {
+				this.skip('No echo server available');
+			}
+			return xhrRequest(getRequestUrl('foo.json'), { timeout: 1 })
 				.then(
-					dfd.resolve.bind(dfd),
-					dfd.callback(function (error: Error): void {
+					null,
+					function (error: Error) {
 						assert.strictEqual(error.name, 'RequestTimeoutError');
-					})
+					}
 				);
 		},
 
-		'user and password'(): void {
-			const dfd = this.async();
-			xhrRequest(getRequestUrl('foo.json'), {
+		'user and password'() {
+			if (!echoServerAvailable) {
+				this.skip('No echo server available');
+			}
+			return xhrRequest(getRequestUrl('foo.json'), {
 				user: 'user',
 				password: 'password'
 			}).then(
-				dfd.callback(function (response: any): void {
+				function (response: any) {
 					assert.strictEqual(response.requestOptions.user, 'user');
 					assert.strictEqual(response.requestOptions.password, 'password');
-				}),
-				dfd.reject.bind(dfd)
+				}
 			);
 		},
 
-		'auth, without user or password'(): void {
-			const dfd = this.async();
-			xhrRequest(getRequestUrl('foo.json'), {
+		'auth, without user or password'() {
+			if (!echoServerAvailable) {
+				this.skip('No echo server available');
+			}
+			return xhrRequest(getRequestUrl('foo.json'), {
 				auth: 'user:password'
 			}).then(
-				dfd.callback(function (response: any): void {
+				function (response: any) {
 					assert.strictEqual(response.requestOptions.user, 'user');
 					assert.strictEqual(response.requestOptions.password, 'password');
-				}),
-				dfd.reject.bind(dfd)
+				}
 			);
 		},
 
+		'query': {
+			'.get with query URL and query option string'() {
+				if (!echoServerAvailable) {
+					this.skip('No echo server available');
+				}
+				return xhrRequest('/__echo/xhr?color=blue&numbers=one&numbers=two', {
+					query: new UrlSearchParams({
+						foo: [ 'bar', 'baz' ],
+						thud: 'thonk',
+						xyzzy: '3'
+					}).toString(),
+					handleAs: 'json'
```

Comment:
'json' is explicitly excluded from the response types handled by xhr, unless I'm missing something.